tests/generate_schedule.spec.ts

```diff
@@ -1,19 +1,55 @@
-import supertest from "supertest";
 import {
-    BACKEND_URL,
-    ALGORITHM1_URL,
-    ALGORITHM2_URL,
-    apollo_client,
+    sendGraphQLQuery,
 } from "./test_config";
 
-// Setup supertest for requests to each module
-const request_backend = supertest(BACKEND_URL);
-const request_algorithm1 = supertest(ALGORITHM1_URL);
-const request_algorithm2 = supertest(ALGORITHM2_URL);
-
 describe("Generate base schedule with courses to timeslots and professors to courses assigned", () => {
-    it("should return a valid base schedule", async () => {
-        // TODO: Add the actual tests to generate base schedule
-        expect(true).toEqual(true);
+    it("should return an authentication error when not logged in", async () => {
+        // Attempt to generate schedule
+        const responseGenerateSchedule = await sendGraphQLQuery(
+          `mutation {
+            generateSchedule(input: { year: 2022 }) {
+                message
+                success
+            }
+          }`
+        );
+
+        const responseJSONLogin = await responseGenerateSchedule.json();
+        expect(responseJSONLogin.data.generateSchedule.message).toEqual("Not logged in");
+        expect(responseJSONLogin.data.generateSchedule.success).toEqual(false);
+    });
+
+    it("should return a valid schedule when logged in", async () => {
+        // Login
+        const responseLogin = await sendGraphQLQuery(
+          `mutation {
+            login(username: "testadmin", password: "testpassword") {
```

Comment:
Out of the loop here, where are these `testadmin` values being set? Is that just in your backend db migration script?

Reply:
It's the default credentials from our db seeding operation.